### PR TITLE
kernel: pre-28.x chainparams and headerssync update

### PR DESCRIPTION
**Pull request by achow101**
Update chainparams and headerssync parameters for the pre-28.x branching, per https://github.com/bitcoin/bitcoin/blob/master/doc/release-process.md#before-branch-off

### CODE REVIEW

File: src/kernel/chainparams.cpp [hunk 1]
```diff
@@ -250,7 +250,7 @@ class CTestNetParams : public CChainParams {
         nDefaultPort = 18333;
         nPruneAfterHeight = 1000;
         m_assumed_blockchain_size = 42;
-        m_assumed_chain_state_size = 3;
+        m_assumed_chain_state_size = 11;
```
Comment (Sjors) [inline src/kernel/chainparams.cpp:253]: 985d1ec267f44b7897855fd2312da76b3fcd165b: this is already 13G for me.

**Review by Sjors — APPROVED** [review]
ACK cd913de6488d25057e3bf7dcad1508e9d689f87f

If you have to touch, can you make `contrib/devtools/headerssync-params.py` executable?

File: src/kernel/chainparams.cpp (hunk 1 above)
Comment (achow101) [reply src/kernel/chainparams.cpp:253]: It's 9.9G for me. Are you looking at the right network datadir?

Comment (mzumsande) [reply src/kernel/chainparams.cpp:253]: I see similar size to @Sjors, as of height=2873716:
```
du -sh ~/.bitcoin/testnet3/chainstate/ ~/.bitcoin/testnet3/blocks/
13G	/home/martin/.bitcoin/testnet3/chainstate/
82G	/home/martin/.bitcoin/testnet3/blocks/
```
(I wasn't continuously online during the fork shenanigans, so having fewer blocks makes sense).

Comment (fjahr) [reply src/kernel/chainparams.cpp:253]: Either way, `m_assumed_blockchain_size` can be updated here too.

Comment (0xB10C) [reply src/kernel/chainparams.cpp:253]:
```
$ du -sh blocks/ chainstate/
84G	blocks/
13G	chainstate/
```

Comment (achow101) [reply src/kernel/chainparams.cpp]: I'm quite confused as to how all of you have way larger blocks and chainstate dirs than I do. I checked my 3 nodes and they all agree on these sizes, and all 3 have very high uptimes.

Anyways, bumped `m_assumed_blockchain_size` to 93 and `m_assumed_chain_state_size` to 19, which I think should cover all sizes given here.

**Comment by achow101** [comment]
> If you have to touch, can you make `contrib/devtools/headerssync-params.py` executable?

Done

**Comment by fjahr** [comment]
re-ACK 221809b81cfcecb04050915eebacffda2599da42

Only changes since last review were bumping the Testnet3 size numbers and making `headerssync-params.py` executable.

https://github.com/bitcoin/bitcoin/compare/cd913de6488d25057e3bf7dcad1508e9d689f87f..221809b81cfcecb04050915eebacffda2599da42

**Comment by Sjors** [comment]
re-ACK 221809b81cfcecb04050915eebacffda2599da42

**Review by glozow — COMMENTED** [review]
ACK 221809b81cfcecb04050915eebacffda2599da42

Checked chain params against my nodes (all except testnet3, I just sanity checked that against a block explorer). Checked I get the same headerssync params.

**Review by marcofleon — COMMENTED** [review]
ACK 221809b81cfcecb04050915eebacffda2599da42

Headerssync params lgtm, got the same ones. I checked the main chain params against my node and the only thing is that my blocks dir is 675 gbs, not 620. Not sure if it matters too much.

**Review by tdb3 — COMMENTED** [review]
post merge ACK 221809b81cfcecb04050915eebacffda2599da42

Checked chainparams (mainnet and signet) manually on each of my nodes.  Matched.

```
$ bitcoin-cli getblockhash 856760
$ bitcoin-cli getchaintxstats 4096 <block hash from getblockhash>
$ bitcoin-cli getblockheader <block hash from getblockhash>
$ bitcoin-cli -signet getblockhash 208800
$ bitcoin-cli -signet getchaintxstats 4096 <block hash from getblockhash>
$ bitcoin-cli -signet getblockheader <block hash from getblockhash>
```

**Pull request merged**